util-app/src/main/scala/com/twitter/app/GlobalFlag.scala

```diff
@@ -116,29 +116,35 @@ abstract class GlobalFlag[T] private[app] (
 
 object GlobalFlag {
 
-  private[app] def get(f: String): Option[Flag[_]] = {
-    def validMethod(m: Method): Boolean =
-      m != null &&
-        Modifier.isStatic(m.getModifiers) &&
-        m.getReturnType == classOf[Flag[_]] &&
-        m.getParameterCount == 0
-
-    def tryMethod(clsName: String, methodName: String): Option[Flag[_]] =
+  private[app] def get(flagName: String): Option[Flag[_]] = {
+    def tryMethod(className: String, methodName: String): Option[Flag[_]] =
       try {
-        val cls = Class.forName(clsName)
+        val cls = Class.forName(className)
         val m = cls.getMethod(methodName)
-        if (validMethod(m))
-          Some(m.invoke(null).asInstanceOf[Flag[_]])
-        else
-          None
+        val isValid = Modifier.isStatic(m.getModifiers) &&
```

Comment:
why did we get rid of the `m != null` check?

Reply:
`Class.getMethod` doesn't return null, there is a null check inside method, and it throws exception, that is catched below.